.github/workflows/build_and_test.yml

```diff
@@ -682,50 +683,12 @@ jobs:
             })
 
   trigger-e2e-tests:
+    if: ${{ (!github.event.pull_request.draft || contains( github.event.pull_request.labels.*.name, 'run-e2e-test-in-draft')) && !cancelled() }}
```

Comment:
Why do we need `!cancelled()` part here?

Reply:
The reason to add canceled was to force run this job if the earlier job are skipped . May be let me check if we can add only skipped for all job

Reply:
to check the skip status for each job it will be a long list,  this seems to be a better alternative to ignore all the skipped job

Reply:
added -${{ head_ref }} which should be unique across branches

Reply:
> The reason to add canceled was to force run this job if the earlier job are skipped

But why do we need to change this logic? Where do we get new skipped jobs (in comparison with the current logic)?
Isn't this some leftovers from the previous approach again?

Reply:
yep true ... that's very silly of me. Not sure why I got stuck in that loop. 

I have removed the 'cancelled' check